generator/overrides/gen.go

```diff
@@ -0,0 +1,435 @@
+					jsonTag := field.Tag.Get("json")
+					if jsonTag != "" &&
+						jsonTag != "-" &&
+						!strings.Contains(jsonTag, ",inline") &&
+						!strings.Contains(jsonTag, ",omitempty") {
+						newJSONTag := jsonTag + ",omitempty"
+						astField.Tag.Value = strings.Replace(astField.Tag.Value, `json:"`+jsonTag+`"`, `json:"`+newJSONTag+`"`, 1)
+						astField.Doc = updateComments(
+							astField, astField.Doc,
+							`.*`,
+							` *`+regexp.QuoteMeta("+optional")+`.*`,
+							` +optional`,
+						)
+
+					}
+				}
+
+				processFieldType := func(ident *ast.Ident) *typeToProcess {
+					typeToOverride, existsInPackage := g.packageTypes[ident.Name]
+					if !existsInPackage {
+						return nil
+					}
+					ident.Name = ident.Name + g.suffix
+					return &typeToProcess{
+						OverrideTypeName: ident.Name,
+						TypeInfo:         typeToOverride,
+					}
+				}
+
+				var fieldTypeToProcess *typeToProcess
+
+				switch fieldType := astField.Type.(type) {
+				case *ast.ArrayType:
+					switch elementType := fieldType.Elt.(type) {
+					case *ast.Ident:
+						fieldTypeToProcess = processFieldType(elementType)
+						if fieldTypeToProcess != nil {
+							fieldTypeToProcess.MandatoryKey = strings.Title(genutils.GetPatchMergeKey(&field))
+						}
+					}
+				case *ast.Ident:
+					fieldTypeToProcess = processFieldType(fieldType)
+					if field.Markers.Get(genutils.UnionDiscriminatorMarker.Name) != nil {
+						enumValues := []string{}
+						for _, f := range typeToOverride.Fields {
+							pos := f.RawField.Pos()
+							fieldChange := fieldChanges[f.RawField.Pos()]
+							if pos != cursor.Node().Pos() &&
+								!shouldSkip(fieldChange.OverrideMarker) {
+								enumValues = append(enumValues, fieldChange.FieldInfo.Name)
+							}
+						}
+						kubebuilderAnnotation := "+kubebuilder:validation:Enum=" + strings.Join(enumValues, ";")
+						astField.Doc = updateComments(astField, astField.Doc, `.*`, ` *`+regexp.QuoteMeta("+kubebuilder:validation:Enum=")+`.*`, kubebuilderAnnotation)
+						fieldTypeToProcess.DropEnumAnnotation = true
+					}
+				case *ast.StarExpr:
+					switch elementType := fieldType.X.(type) {
+					case *ast.Ident:
+						fieldTypeToProcess = processFieldType(elementType)
+					}
+				case *ast.MapType:
+					switch elementType := fieldType.Key.(type) {
+					case *ast.Ident:
+						fieldTypeToProcess = processFieldType(elementType)
+					}
+					switch elementType := fieldType.Value.(type) {
+					case *ast.Ident:
+						fieldTypeToProcess = processFieldType(elementType)
+					}
+				default:
+				}
+
+				if fieldTypeToProcess != nil {
+					moreTypesToAdd = append(moreTypesToAdd, *fieldTypeToProcess)
+				}
+			}
+			return true
+		},
+		func(*astutil.Cursor) bool { return true },
+	).(*ast.GenDecl)
+
+	if len(errors) > 0 {
+		return nil, moreTypesToAdd, errors // Check there is only 1 typespec in it or trigger an error
```

Comment:
Is this commit still needed? Looks like this is checked above.

Reply:
Fixed in commit https://github.com/devfile/api/pull/133/commits/37c1e341bb7bb541566dbf0b63a536b64e1b6028

Reply:
`s/commit/comment/`, of course -- woops :)